homeassistant/components/google_sheets/__init__.py

```diff
@@ -52,18 +52,31 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except aiohttp.ClientError as err:
         raise ConfigEntryNotReady from err
 
-    if not async_entry_has_scopes(hass, entry):
+    if not async_entry_has_scopes(entry):
         raise ConfigEntryAuthFailed("Required scopes are not present, reauth required")
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = session
 
     await async_setup_service(hass)
 
+    entry.async_on_unload(entry.add_update_listener(async_reload_entry))
+
     return True
 
 
-def async_entry_has_scopes(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+def get_feature_access(config_entry: ConfigEntry) -> list[str]:
+    """Return the desired sheets feature access."""
+    return [
+        FeatureAccess.file.value,
+        FeatureAccess[config_entry.options[CONF_SHEETS_ACCESS]].value,
```

Comment:
The read scope gives read access to all spreadsheets. The file scope ensures users still have write access to the document that the integration created. I plan on extending the append service to allow the user to add a document id so they can specify a different document other than the integration made one.

The plan is to add an edit service which makes more sense for any other documents that people have access to that they want to modify. It should be the same schema for both services.

Reply:
Yeah I understand the scope differences, just wanting to see how they connect to the feature to ensure they all make sense together.

Ok, I'd say let's do the scope changes with the features it will enable? I don't mind viewing the larger PR. I just want to make sure it all connects.